### DISCUSSION TITLE
Ignore indentation style

### DISCUSSION THREAD

**Discussion by ShackGH** [astral-sh/ruff#11004] [Q&A] [answered]
The current project I'm working on has some files which are indented by tabs and some that are indented by spaces. Is there a way to ignore indentation as a whole or enable both tabs or spaces (not in same file)?

**Comment by zanieb** [comment, marked as answer]
Ruff supports hierarchical configuration, so if you can isolate the files into directories grouped by space or tab indentation you can override the configuration per directory. Otherwise, I think you're kind of out of luck. It seems bad to have mixed standards on the project though?

**Reply by ShackGH** [reply]
Thanks for the answer. The project is mostly written in other languages but they have a couple of services that are run in python, and the python files doesn't seem to have a uniform indentation style unfortunately.

**Reply by MichaReiser** [reply]
Where is the mixed indentation style a problem when using Ruff? Like what commands are you running and what "errors" are you seeing?

I'm asking because it's true that enforcing a consistent indentation is very much intended when using ruff's formatter, but it should be okay to use mixed indentation when using Ruff's linter, but you might need to disable a few rules.

**Reply by ShackGH** [reply]
I have a .toml file with the default configurations (default indentation style is space). And if I run `ruff format` ,it would format a number files that have tab indentation and if I change the indentation style to `tabs` in the .toml, then the files that have spaces would get formatted.

**Reply by MichaReiser** [reply]
Yeah, there's nothing you can do about `ruff format`. The purpose of `format` is a consistent formatting and that includes indentation. But I suspect that the formatter doesn't just change indentation. Have you considered just reformatting the code once in a single commit? You can then ["hide" the commit from the blame view](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

**Reply by ShackGH** [reply]
Haven't considered it, will take a look. Thanks.
